### ISSUE TITLE
Distinguish between map and property access in expressions

### ISSUE THREAD

**Issue by yjbanov** [angular/angular#420]
This has been an issue in AngularDart 1.0:

```
@Component(...)
class Foo implements Map {
  String bar;
}
```

```
<div>{{foo.bar}}</div>
```

This template would not reach the `bar` getter and instead try to get the value via `foo['bar']`. This is because AngularDart checks if `foo is Map` and ignores the getter. Angular 2 should have separate syntax for map access that makes it explicit in the expression whether map access or property access is used. Example:

```
{{foo.bar}}
{{foo:bar}}
```

**Comment by mhevery** [comment]
Closing as this should already be the case in NG2

**Comment by yjbanov** [comment]
Reopening the issue as we do not have tests or docs for this feature.

**Comment by yjbanov** [comment]
Fixed by #2073
